sixMans/game.py

```diff
@@ -192,9 +257,213 @@ async def process_captains_pick(self, reaction, user):
             for player in self.orange:
                 await self.add_to_orange(player)
             self.reset_players()
+            await self.update_game_info()
         
-        await self._notify(new_state="ongoing")
         return teams_complete
+    
+    async def process_team_select_vote(self, reaction, member, added=True):
+        if member not in self.players:
+            return
+
+        if self._hex_i_from_emoji(reaction.emoji) not in SELECTION_MODES:
+            return 
+        
+        # COUNT UP VOTE TOTALS
+        if not self.vote:
+            self.vote = [None, 0]
+
+        # RECORD VOTES
+        self.info_message = await self.textChannel.fetch_message(self.info_message.id)
+        votes = {}
+        for this_react in self.info_message.reactions:
+            # here maybe
+            react_hex_i = self._hex_i_from_emoji(this_react.emoji)
+            if react_hex_i in SELECTION_MODES:
+                reacted_members = await this_react.users().flatten()
+                count = this_react.count - 1
+                if added and this_react.emoji != reaction.emoji and member in reacted_members:
+                    await this_react.remove(member)
+                    count -= 1
+                votes[react_hex_i] = {'count': count, 'emoji': this_react.emoji}
+
+        # Update embed
+        embed = self._get_vote_embed(votes)
+        await self.info_message.edit(embed=embed)
+
+        # COUNT VOTES - Check if complete
+        total_votes = 0
+        runner_up = 0
+        running_vote = [None, 0]
+
+        for react_hex, data in votes.items():
+            num_votes = data['count']
+            if num_votes > running_vote[1]:
+                runner_up = running_vote[1]
+                running_vote = [react_hex, num_votes]
+
+            elif num_votes > runner_up and num_votes <= running_vote[1]:
+                runner_up = num_votes
+
+            total_votes += num_votes
+        
+        # track top vote
+        if running_vote[1] > self.vote[1]:
+            self.vote = running_vote
+        
+        pending_votes = len(self.players) - total_votes
+
+        voted_mode = None
+        # Vote Complete if...
+        # here
+        # if added and member.name == 'nullidea' or (pending_votes == 0 or (pending_votes + runner_up) <= self.vote[1]):
+        if added and (pending_votes == 0 or (pending_votes + runner_up) <= self.vote[1]):
+            # action and update first - help with race conditions
+            voted_mode = SELECTION_MODES[self.vote[0]]
+            if self.teamSelection.lower() == Strings.VOTE_TS.lower():
+                self.teamSelection = voted_mode
+                # await self.textChannel.send(self.teamSelection)
```

Comment:
Is this a left over comment that should be removed?

Reply:
Yes